addons/mail_plugin/models/res_partner_iap.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
+
+from odoo import fields, models
+
+
+class ResPartnerIap(models.Model):
+    """Technical model which stores the response returned by IAP.
+
+    The goal of this model is to not enrich 2 times the same company. We do it in a
+    separate model to not add heavy field (iap_enrich_info) on the <res.partner>
+    model.
+
+    We also save the requested domain, so whatever the values are on the <res.partner>,
+    we will always retrieve the already enriched <res.partner> and the corresponding
+    IAP information.
+    """
+
+    _name = 'res.partner.iap'
+    _description = 'Partner IAP'
+
+    partner_id = fields.Many2one('res.partner', string='Partner', help='Corresponding partner',
```

Comment:
I would add an unique constraint on it

Reply:
Maybe a unique index for clarity, since we're probably going to look partner.iap records by partner_id a lot?

Reply:
Well an unique constraint implies an index at db level, am I wrong ?

Reply:
@xmo-odoo What do you means by "unique index" ? I already added the unique constraint like this
```
 _sql_constraints = [('unique_partner_id', 'UNIQUE(partner_id)', 'Only one partner IAP is allowed for one partner')]
```

Reply:
unique constraint imply  an index on the DB, but index=True does not imply a unique constraint, so we can not just add "index=True" on the field if it's what you wanted to do